src/walletSdk/Anchor/index.ts

```diff
@@ -195,6 +218,183 @@ export class Anchor {
     }
   }
 
+  /**
+  * Watch all transactions returned from a transfer server. When new or
+  * updated transactions come in, run an `onMessage` callback.
+  *
+  * On initial load, it'll return ALL pending transactions via onMessage.
+  * Subsequent messages will be any one of these events:
+  *  * Any new transaction appears
+  *  * Any of the initial pending transactions change any state
+  *
+  * You may also provide an array of transaction ids, `watchlist`, and this
+  * watcher will always react to transactions whose ids are in the watchlist.
+  */
+  watchAllTransactions(params: {
+    authToken: string;
+    assetCode: string;
+    onMessage: (transaction) => void;
+    onError: (error: any) => void;
+    watchlist?: string[];
+    timeout?: number;
+    isRetry?: boolean;
+    noOlderThan?: string;
+    kind?: string;
+    lang?: string;
+  }): WatcherResponse {
+    const {
+      authToken,
+      assetCode,
+      onMessage,
+      onError,
+      watchlist = [],
+      timeout = 5000,
+      isRetry = false,
+      ...otherParams
+    } = params;
+    
+    // make an object map out of watchlist
+    const watchlistMap: any = watchlist.reduce(
+      (memo: any, id: string) => ({ ...memo, [id]: true }),
+      {},
+    );
+
+    // make sure to initiate registries for the given asset code
+    // to prevent 'Cannot read properties of undefined' errors
+    if(!this._transactionsRegistry[assetCode]) {
+      this._transactionsRegistry[assetCode] = {};
+    }
+    if(!this._transactionsIgnoredRegistry[assetCode]) {
+      this._transactionsIgnoredRegistry[assetCode] = {};
+    }
+
+    // if it's a first run, drop it in the registry for the given asset code
+    if (!isRetry) {
+      this._watchAllTransactionsRegistry[assetCode] = true;
+    }
+
+    this.getTransactionsForAsset({ authToken, assetCode, ...(otherParams || {}) })
```

Comment:
we can just have it as just this right?
```suggestion
    this.getTransactionsForAsset({ authToken, assetCode, ...otherParams })
```

Reply:
Nice catch! I thought `otherParams` could become `undefined` in case no optional params were provided but I just discovered it would actually become `{}` in that case :)